src/launchpads/MK2/LaunchpadMK2.ts

```diff
@@ -63,6 +63,51 @@ export class LaunchpadMK2 extends BaseLaunchpad {
     this.sendSysEx(40, 0, buttonMapped, validatePaletteColor(color));
   }
 
+  /**
+   * @inheritDoc
+   */
+  public setButtons(...buttons: ButtonStyle[]): void {
+    // For the MK2, we can set multiple buttons at once per style command, so
+    // we first need to group by style.
+    const groups = groupByStyle(buttons);
+
+    // Combine 'off' and 'palette', since 'off' is just an alias for palette 0
+    groups.palette.push(...groups.off.map(s => ({
+      button: s.button,
+      style: {
+        style: 'palette',
+        color: 0
+      } as const,
+    })));
+
+    if (groups.palette.length > 0) {
+      this.sendSysEx(10, ...groups.palette.flatMap(s => [
+        this.mapButtonFromXy(s.button),
+        validatePaletteColor(s.style.color),
+      ]));
+    }
+    if (groups.rgb.length > 0) {
+      this.sendSysEx(11, ...groups.rgb.flatMap(s => [
+        this.mapButtonFromXy(s.button),
+        ...scaleRgbMk2(s.style.rgb),
+      ]));
+    }
+    if (groups.flash.length > 0) {
+      this.sendSysEx(35, ...groups.flash.flatMap(s => [
+        0, // Is this 0 repeated for every button? Guide indicates yes but I can't test
```

Comment:
I'll test this before merging

Reply:
Is this the desired result on the MK2?

https://user-images.githubusercontent.com/8843386/162939388-2cd0cb9e-7ea5-4680-90ba-1712422c1fce.mp4

Reply:
Yes!